index.bs

```diff
@@ -221,6 +221,24 @@ and doing cleanup as necessary (see the [[#listen-fully-active]] pattern above).
   but every feature specification should think about this and decide what works best for the feature.
 </div>
 
+<h4 id="non-synchronous-steps">Be aware that navigations can happen between steps running [=in parallel=]</h4>
+If steps that run [=in parallel=] queue a task that is associated with a document,
+you need to be aware that the document might have been navigated away from and (later on) navigated to again.
+This can happen while running the steps [=in parallel=] and can have happened when the steps in the queued task eventually run.
+Since a navigation has happened and a considerable amount of time might have passed between the steps,
+the user might not expect some states that are saved in the steps to be preserved.
+
+When writing steps that run [=in parallel=],
+consider also listening to changes to the [=Document/fully active=] state
+and doing cleanup as necessary or even refrain from running some steps
+(see the [[#listen-fully-active]] pattern above).
+<div class=example>
+  APIs that request permission to the user, or requests user interaction through prompt, etc.,
+  should consider re-requesting the user if it detects that the document had become non-fully active.
+</div>
```

Comment:
I wish we had a better example here. I don't think this is particularly actionable for spec authors, e.g., I don't see how the advice here would change how https://w3c.github.io/permissions/#dfn-request-permission-to-use is written.

Note that because that algorithm queues a task, and tasks don't run on non-fully active documents, the result would be that the permissions store is updated after being reactivated from bfcache. That seems OK?

@annevk, did you have other examples of problematic cases you've seen in various specs?

Reply:
Well, there's no guarantee the document is still fully active when "ask the user" happens, right? I think that's the potentially more problematic aspect.

Reply:
I guess I'm just still unsure on what the concrete guidance is. Since @annevk is the editor of the Storage Standard, does he have suggestions on how that algorithm should be updated to handle this concern?

Reply:
Yeah, so, I feel like if neither you nor I can figure out how to apply the advice here to any existing specs, this advice isn't going to be that useful for other spec authors... So I wonder why we're adding it.

Reply:
That is one way of approaching it. Another way of approaching it is whether we agree there is a problem to be solved here.